zerver/lib/markdown/__init__.py

```diff
@@ -574,14 +574,21 @@ def get_tweet_id(url: str) -> Optional[str]:
     return tweet_id_match.group("tweetid")
 
 
-class InlineHttpsProcessor(markdown.treeprocessors.Treeprocessor):
+class InlineImageProcessor(markdown.treeprocessors.Treeprocessor):
     def run(self, root: Element) -> None:
         # Get all URLs from the blob
         found_imgs = walk_tree(root, lambda e: e if e.tag == "img" else None)
         for img in found_imgs:
             url = img.get("src")
             assert url is not None
-            if urllib.parse.urlsplit(url).scheme != "http":
+            split_url = urllib.parse.urlsplit(url)
+            if (
+                f"{split_url.scheme}://" == settings.EXTERNAL_URI_SCHEME
+                and (
+                    f".{split_url.netloc}".endswith(f".{settings.EXTERNAL_HOST}")
+                    or split_url.netloc in settings.REALM_HOSTS.values()
+                )
+            ) or (split_url.scheme == "" and split_url.netloc == ""):
                 # Don't rewrite images on our own site (e.g. emoji).
                 continue
```

Comment:
What do you think about moving this block of code to a function in `zerver/lib/subdomains.py` with a name like `url_is_hosted_by_installation(url)`, and perhaps writing a small test suite for it (while it's nice that this has end-to-end tests via the markdown processor, it seems like we could write nicely readable 1-line tests for this more readily with a boolean return value).

I'd also like to see a block comment explaining the policy being implemented, maybe just a docstring for `InlineImageProcessor`, and an edit of `docs/production/security-model.md` to update this:
```
* Zulip supports using the Camo image proxy to proxy content like                                    
  inline image previews that can be inserted into the Zulip message                                  
  feed by other users over HTTPS.                                                                    
```